### PR TITLE
Feature/321 tech build for basket

### PR DESCRIPTION
**Pull request by jenbutongit**
DB is plugged in to required learning for the time being. Planned learning is just hard coded for now

### CODE REVIEW

File: src/service/ui/page/profile/edit.html [hunk 1]
```diff
@@ -28,16 +28,27 @@ <h2 class="heading-medium error-summary-heading" id="error-summary-heading-examp
...
+                        {{/if}}
+                        <input class="form-control" id="givenName" name="givenName" type="text" value="{{user.givenName || ''}}">
+                    </div>
+
 
                     <div class="form-group {{validFields == false && !user.department ?  "form-group-error": ''}}">
                         <label class="form-label" for="department">Department</label>
                         {{#if validFields == false && !user.department}}
-                        <span class="error-message">Department field cannot be empty.</span>
+                        <span class="error-message">Department field cwwwannot be empty.</span>
```
Comment (tav) [inline src/service/ui/page/profile/edit.html]: Is the `cwwwannot` intentional?

File: src/service/ui/src/controllers/learning-record/index.ts [hunk 2]
```diff
@@ -48,7 +48,12 @@ export async function courseResult(req: Request, res: Response) {
 	}
 }
 
-async function getLearningRecord(user: any) {
+export enum CourseState {
```
Comment (tav) [inline src/service/ui/src/controllers/learning-record/index.ts]: Yay! Enums! :)

File: src/service/ui/src/controllers/user.ts [hunk 3]
```diff
@@ -120,9 +136,9 @@ export function updateProfile(req: Request, res: Response) {
 		uri: config.AUTHENTICATION.serviceUrl + '/scim2/Users/' + req.user.id,
 	}
 
-	request(options, (error, response, body) => {
-		if (!error && response.statusCode === 200) {
-			updateUserObject(req, updateProfileObject.CshrUser)
+	request(options, (error: Error, response: Response, body: Body) => {
```
Comment (tav) [inline src/service/ui/src/controllers/user.ts]: Any reason for adding these type annotations here? I think the `Response` we have here is `express.Response` which isn't what `request` is expecting?

File: src/service/ui/page/profile/edit.html (hunk 1 above)
Comment (superafroman) [inline src/service/ui/page/profile/edit.html]: 😂

File: src/service/ui/src/controllers/user.ts [hunk 4]
```diff
@@ -120,9 +136,9 @@ export function updateProfile(req: Request, res: Response) {
...
+		if (!error && response.statusCode == 200) {
```
Comment (tav) [inline src/service/ui/src/controllers/user.ts]: Although, in this case, the `==` should be fine, doesn't the linter enforce triple equals everywhere?

File: src/service/ui/src/controllers/user.ts [hunk 5]
```diff
@@ -84,6 +87,14 @@ export function signIn(req: Request, res: Response) {
...
 	}
 }
+export interface User {
+	id: string
+	email: string
+	givenName: string
+	department: string
+	profession: string
+	grade: string
+}
```
Comment (superafroman) [inline src/service/ui/src/controllers/user.ts]: User is now a class in ./src/lib/model/user

File: src/service/ui/src/controllers/user.ts (hunk 3 above)
Comment (superafroman) [inline src/service/ui/src/controllers/user.ts]: Update to use axios (I can help if needed)

File: src/service/ui/page/profile/edit.html (hunk 1 above)
Comment (jenbutongit) [reply src/service/ui/page/profile/edit.html]: no wonder nothing closed when I tried cmd+w :)

File: src/service/ui/src/controllers/basket.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import {Request, Response} from 'express'
+import * as template from 'lib/ui/template'
+import * as learningRecord from './learning-record'
+
+export let basketPage = async (req: Request, res: Response) => {
```
Comment (tav) [inline src/service/ui/src/controllers/basket.ts]: For future, make toplevel a function declaration as opposed to an arrow function.

File: src/service/ui/src/controllers/user.ts [hunk 7]
```diff
@@ -98,39 +102,49 @@ export function tryUpdateProfile(req: Request, res: Response) {
...
-		headers: {'Content-Type': 'application/json'},
-		method: 'PUT',
-		rejectUnauthorized: false, //Jen - TODO: Is there a securer way to do this?
-		uri: config.AUTHENTICATION.serviceUrl + '/scim2/Users/' + req.user.id,
 	}
-
-	request(options, (error, response, body) => {
-		if (!error && response.statusCode === 200) {
-			updateUserObject(req, updateProfileObject.CshrUser)
+	Axios.request(requestConfig)
```
Comment (tav) [inline src/service/ui/src/controllers/user.ts]: For future, use `axios.put`?

**Pull request merged**